scene/main/viewport.h

```diff
@@ -271,7 +272,10 @@ class Viewport : public Node {
 	bool handle_input_locally = true;
 	bool local_input_handled = false;
 
+	// Collider to frame
 	Map<ObjectID, uint64_t> physics_2d_mouseover;
+	// Collider & shape to frame
+	Map<Pair<ObjectID, int>, uint64_t, PairSort<ObjectID, int>> physics_2d_shape_mouseover;
```

Comment:
Why is `PairSort<ObjectID, int>` needed? Does it fail to compile otherwise?

Reply:
Yes. You need to define a comparator operator for Map to know how to hash the objects.

Reply:
Ah yes, that makes sense.